plugins/optimization-detective/detect.js

```diff
@@ -161,6 +161,17 @@ async function getAlreadySubmittedSessionStorageKey(
 	currentUrl,
 	urlMetricGroupStatus
 ) {
+	// Check if crypto.subtle is available.
+	if ( ! window.crypto || ! window.crypto.subtle ) {
+		// eslint-disable-next-line no-console
+		console.warn(
+			'[Optimization Detective] Web Crypto API is not available. This API is only available in secure contexts (HTTPS). Detection cannot proceed. If you are testing locally, ensure you use HTTPS or run on localhost.'
+		);
+		throw new Error(
+			'Web Crypto API is unavailable in this context. Try using HTTPS or localhost.'
+		);
```

Comment:
Shouldn't this reuse the logger?

```suggestion
		error(
			'[Optimization Detective] Web Crypto API is not available. This API is only available in secure contexts (HTTPS). Detection cannot proceed. If you are testing locally, ensure you use HTTPS or run on localhost.'
		);
		return;
```